src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,221 @@
+package edu.berkeley.ce.rockslicing
+
+import breeze.linalg
+import breeze.linalg.DenseVector
+import scala.annotation.tailrec
+
+object JointGenerator {
+  /**
+    * Finds the normal to the joint specified by the input strike and dip
+    *
+    * @param strike Strike of the joint plane given in degrees
+    * @param dip Dip of the joint plane given in degrees
+    * @return Array that specifies the x, y and z components of the joint plane
+    */
+  private def findJointNormal(strike: Double, dip: Double): Array[Double] = {
+    val strikeRadians = strike.toRadians
+    val dipRadians = dip.toRadians
+    val strikeVector = DenseVector[Double](math.cos(-strikeRadians), math.sin(-strikeRadians), 0.0)
+    // If joint is vertical, dip vector will point along negative z-axis
+    if (math.abs(dip - 90.0) < NumericUtils.EPSILON) {
```

Comment:
How about you do something more like
```scala
val dipVector = if (math.abs(dip - 90.0) < NumericUtils.EPSILON) {
    DenseVector[Double](0.0, 0.0, -1.0)
} else {
    DenseVector[Double](math.cos(dipRadians)...
}
val jointNormal = ...
Array(...
````

This allows you to avoid a bit of duplicate code.